ci-operator/step-registry/baremetalds/ofcir/setup/baremetalds-ofcir-setup-commands.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+set -x
+
+echo "************ baremetalds ofcir setup command ************"
+
+function send_slack(){
+    set +x
+    SLACK_AUTH_TOKEN="T027F3GAJ/B011TAG710V/$(cat $CLUSTER_PROFILE_DIR/slackhook)"
+
+    sleep infinity
+    curl -X POST --data "payload={\"text\":\"<https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/$JOB_NAME/$BUILD_ID|Ofcir setup failed> $1\n\"}" \
+        "https://hooks.slack.com/services/${SLACK_AUTH_TOKEN}"
+    set -x
+}
+
+function exit_with_success(){
+  cat >"${ARTIFACT_DIR}/junit_metal_setup.xml" <<EOF
+  <testsuite name="metal infra" tests="1" failures="0">
+    <testcase name="[sig-metal] should get working host from infra provider"/>
+  </testsuite>
+EOF
+  exit 0
+}
+
+function exit_with_failure(){
+  # TODO: update message to reflect job name/link
+  MESSAGE="baremetalds: ${1:-"Failed to create ci resource: ipi-${NAMESPACE}-${JOB_NAME_HASH}-${BUILD_ID}"}"
+  echo $MESSAGE
+  cat >"${ARTIFACT_DIR}/junit_metal_setup.xml" <<EOF
+  <testsuite name="metal infra" tests="1" failures="1">
+    <testcase name="[sig-metal] should get working host from infra provider">
+      <failure message="">$MESSAGE</failure>
+   </testcase>
+  </testsuite>
+EOF
+  send_slack "$MESSAGE"
+  exit 1
+}
+
+trap 'exit_with_failure' ERR
+
+cd
+
+cat > "${SHARED_DIR}/packet-conf.sh" <<-EOF
```

Comment:
nit: it could be useful to comment this section about it's still really nee: the name of the generated script, `packet-conf.sh` could say a little bit weird in the current context